### PR TITLE
Add biosppy.ml sub-package with AFib detection model

### PR DESCRIPTION
**Pull request by rafaelscsilva**
This PR introduces the `biosppy.ml` sub-package, a machine-learning extension for biosignals. The first module, `ecg.py`, includes an `AFibDetection` model, a pre-trained bidirectional LSTM that detects atrial fibrillation from an RR interval sequence. 

The sub-package is fully **optional**: `tensorflow` is declared as extras in `setup.py` and is never imported unless the user explicitly uses the sub-package.

## Installation

```bash
# BioSPPy ML
pip install biosppy[ml]
```

## Architecture

```
biosppy/ml/
├── __init__.py
├── utils_ml.py          # KerasClassifier, the abstract class for all models
├── ecg_ml.py            # ECG-related models (AFibDetection)
└── _models/          # Pre-trained model files (private)
    ├── ecg_afibdetection_bilstm.h5
    └── ecg_afibdetection_bilstm_details.json
```

### `KerasClassifier` (base class, `utils_ml.py`)

Abstract class that any future Keras-based classification model inherits from. It:
1. Validates that all file paths exist
2. Loads the `.h5` model with `keras.models.load_model`
3. Reads `.json` details file and sets its keys as instance attributes. This is how hyperparameters (`win_len`, `step`, `threshold`) and metadata (`label_map`, `sampling_rate`) are configured to the model and preprocessing.
4. Optionally loads a `joblib` scaler for signal/feature normalization

The base `predict()` is meant to be used in a generic case: preprocess → forward pass → threshold/argmax → classification output.
New classification model additions can override the classes `_preprocess_signal()` and `predict()` for signal-specific cases.

*Note: a new class should be created for segmentation.*

### `AFibDetection` (`ecg_ml.py`)

Detects atrial fibrillation from an RR interval sequence using a pre-trained **bidirectional LSTM** [Silva23].

**Pipeline inside `predict()`:**
1. Input validation
2. `_preprocess_signal`: slides a window of `win_len=20` RR intervals with `step=1` across the sequence
3. Reshape to `(n_windows, 20, 1)` for the BiLSTM input
4. Forward pass: one probability per window
5. If **any** window exceeds `threshold=0.485`, return `True` (AFib present)

*Note: Since the model does binary classification, it returns a boolean instead of the label string.*

## Working example

```python
import numpy as np
from biosppy import storage
from biosppy.ml.ecg_ml import AFibDetection

# Load RR intervals (in ms)
rri, _ = storage.load_txt('examples/rri.txt')

# Run AFib detection
model = AFibDetection()
result = model.predict(rri)

print(f"AFib detected: {result}")   # AFib detected: True
```

### `biosppy/ml/_models/`
Contains the model files, including:
- <model_name>.h5: model architecture and weights in Keras format (required)
- <model_name>_details.json: hyperparameters and metadata loaded at runtime (required)
- <model_name>_scaler.joblib: fitted preprocessing scaler (optional)

### `setup.py` + `MANIFEST.in`
- Added `package_data` entry so `_models/*.h5` and `_models/*.json` are included in the wheel 
- Added `recursive-include biosppy/ml/_models *` to `MANIFEST.in` for the source distribution
- `tensorflow` is in `extras_require['ml']`

## References

> [Silva23] R. Silva, L. Abrunhosa Rodrigues, A. Lourenço, H. Plácido da Silva, *"Temporal Dynamics of Drowsiness Detection Using LSTM-Based Models"*, International Work-Conference on Artificial Neural Networks, pp. 211–220, 2023.

### CODE REVIEW

**Comment by rafaelscsilva** [comment]
@sofia3ms I updated the PR by renaming the modules to *_ml.py like you suggested

**Review by sofia3ms — APPROVED** [review]
